content/blog/2018/egress-mongo/index.md

```diff
@@ -0,0 +1,1137 @@
+      subsets:
+        - name: mongo
+          trafficPolicy:
+            loadBalancer:
+              simple: ROUND_ROBIN
+            portLevelSettings:
+            - port:
+                number: 443
+              tls:
+                mode: ISTIO_MUTUAL
+                sni: placeholder.com
+    EOF
+    {{< /text >}}
+
+1.  Route the traffic destined for _*.com_ to the egress gateway and from the egress gateway to the SNI proxy.
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: direct-mongo-through-egress-gateway
+    spec:
+      hosts:
+      - "*.com"
+      gateways:
+      - mesh
+      - istio-egressgateway-with-sni-proxy
+      tls:
+      - match:
+        - gateways:
+          - mesh
+          port: $MONGODB_PORT
+          sni_hosts:
+          - "*.com"
+        route:
+        - destination:
+            host: istio-egressgateway-with-sni-proxy.istio-system.svc.cluster.local
+            subset: mongo
+            port:
+              number: 443
+          weight: 100
+      tcp:
+      - match:
+        - gateways:
+          - istio-egressgateway-with-sni-proxy
+          port: 443
+        route:
+        - destination:
+            host: sni-proxy.local
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application again and verify that the ratings are still displayed correctly.
+
+1.  Check the statistics of the egress gateway's Envoy proxy and see a counter that corresponds to your requests to
+    _*.com_ (the counter for traffic to the SNI proxy). If Istio is deployed in the `istio-system` namespace, the command
+    to print the counter is:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -l istio=egressgateway-with-sni-proxy -n istio-system -o jsonpath='{.items[0].metadata.name}') -c istio-proxy -n istio-system -- curl -s localhost:15000/stats | grep sni-proxy.local.upstream_cx_total
+    cluster.outbound|8443||sni-proxy.local.upstream_cx_total: 1
+    {{< /text >}}
+
+1.  Check the logs of the SNI proxy. If Istio is deployed in the `istio-system` namespace, the command to print the
+    log is:
+
+    {{< text bash >}}
+    $ kubectl logs $(kubectl get pod -l istio=egressgateway-with-sni-proxy -n istio-system -o jsonpath='{.items[0].metadata.name}') -n istio-system -c sni-proxy
+    127.0.0.1 [23/Aug/2018:03:28:18 +0000] TCP [<your MongoDB host>]200 1863 482 0.089
+    127.0.0.1 [23/Aug/2018:03:28:18 +0000] TCP [<your MongoDB host>]200 2590 1248 0.095
+    {{< /text >}}
+
+#### Understanding what happened
+
+In this section you configured egress traffic to your MongoDB host using a wildcarded domain. While for a single MongoDB
+host there is no gain in using wildcarded domains (an exact hostname can be specified), it could be beneficial for the
```

Comment:
s/for the/for/

Reply:
Done in 0e8e7d081d4ff5937d63ee7b698b16c2137edde4.